packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -489,6 +489,7 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
   late List<GlobalKey> buttonItemKeys;
   final MenuController _controller = MenuController();
   late bool _enableFilter;
+  late bool _enableSearch;
   late List<DropdownMenuEntry<T>> filteredEntries;
   List<Widget>? _initialMenu;
   int? currentHighlight;
```

Comment:
One thing that could improve readability here would be to initialize the `late` variables at the declaration site, rather than inside `initState()`.

```dart
late List<GlobalKey> buttonItemKeys = List<GlobalKey>.generate(filteredEntries.length, (int index) => GlobalKey());
final MenuController _controller = MenuController();
late bool _enableFilter = widget.enableFilter;
late bool _enableSearch = widget.enableSearch;
late List<DropdownMenuEntry<T>> filteredEntries = widget.dropdownMenuEntries;
List<Widget>? _initialMenu;
int? currentHighlight;
double? leadingPadding;
late bool _menuHasEnabledItem = filteredEntries.any((DropdownMenuEntry<T> entry) => entry.enabled);
TextEditingController? _localTextEditingController;
```

(Additionally, `_localTextEditingController` could be refactored into a `late` non-nullable field, but that would involve many other changes and might best be saved for its own PR.)